front/components/assistant_builder/InstructionScreen.tsx

```diff
@@ -228,6 +232,7 @@ function AdvancedSettings({
   const supportedModelConfig = getSupportedModelConfig(
     generationSettings.modelSettings
   );
+  const whiteListedProviders = owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
```

Comment:
nit: whiteListedProviders === null => we show all models; whitelistedproviders === [] => we show no models?
blacklistedProviders seemed a tad more natural in this respect (but maybe not appropriate for the use case?)

Reply:
> whiteListedProviders === null => we show all models;

Indeed! However we won't let `whiteListedProviders` to be empty as it should contain at least one provider.

I went for the `whiteListedProviders` option over the `blacklistedProviders` one for the following reason: when we add a new provider we don't want it to be whitelisted by default. Instead we would like an admin to manually whitelist it. 
This behaviour would be a tiny bit more annoying to implement with a blacklist